src/python/pants/base/build_environment.py

```diff
@@ -50,23 +54,46 @@ def is_in_container() -> bool:
     )
 
 
-class _GitIinitialized(Enum):
+class _GitInitialized(Enum):
     NO = 0
 
 
-_Git: _GitIinitialized | Git | None = _GitIinitialized.NO
+_Git: _GitInitialized | GitResult = _GitInitialized.NO
 
 
-def get_git() -> Git | None:
+@dataclasses.dataclass(frozen=True)
+class GitResult(EngineAwareReturnType):
+    git: Git | None
+
+    def cacheable(self) -> bool:
+        return False
```

Comment:
@stuhood are you sure this is what we want? If I understand correctly this only turns off caching of discovering the binary itself. What we really want is to disable caching any time that we run the binary.